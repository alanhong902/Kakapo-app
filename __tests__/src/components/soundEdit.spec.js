import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { map } from 'ramda';
import { notifyActions, soundActions } from 'actions/';
import { SoundEdit } from 'components/';
import { newSoundObj } from 'utils/';
import { getData, createComponentWithIntl, mockEvent } from '../helper';

Enzyme.configure({ adapter: new Adapter() });

function setup(props = {}) {
  const propData = {
    ...getData('themes'),
    soundActions: {},
    ...props,
  };
  const comp = <SoundEdit {...propData} />;
  return {
    props,
    wrapper: shallow(comp),
    tree: createComponentWithIntl(comp).toJSON(),
  };
}

const soundProp = (props = {}) => {
  const obj = { ...newSoundObj, source: 'file', progress: 1, ...props };
  return { sound: map(x => (x === null ? 'wind' : x), obj) };
};

const sound = { ...soundProp() };

test('<SoundEdit/> render', () => {
  const { tree } = setup(sound);
  expect(tree).toMatchSnapshot();
});

test('<SoundEdit/> handleCancel', () => {
  const dispatch = jest.fn();
  const action = soundActions.edit(sound, null);
  const { wrapper } = setup({ sound, dispatch });
  wrapper
    .find('.button')
    .at(0)
    .simulate('click', mockEvent);
  expect(dispatch).toBeCalled();
  expect(dispatch.mock.calls[0][0]).toEqual(action);
});

test('<SoundEdit/> handleSubmit empty', () => {
  const dispatch = jest.fn();
  const action = notifyActions.send('Please fill out all fields');
  const { wrapper } = setup({ sound, dispatch });
  wrapper.find('form').simulate('submit', {
    ...mockEvent,
    target: { getElementsByTagName: () => ({}) },
  });
  expect(dispatch).toBeCalled();
  expect(dispatch.mock.calls[0][0].type).toBe(action.type);
  expect(dispatch.mock.calls[0][0].data).toBe(action.data);
});

test('<SoundEdit/> handleSubmit filled', () => {
  const dispatch = jest.fn();
  const action = soundActions.edit(sound, { name: 'hi' });
  const { wrapper } = setup({ sound, dispatch });
  wrapper.find('form').simulate('submit', {
    ...mockEvent,
    target: { getElementsByTagName: () => [{ name: 'name', value: 'hi' }] },
  });
  expect(dispatch).toBeCalled();
  expect(dispatch.mock.calls[0][0]).toEqual(action);
});
