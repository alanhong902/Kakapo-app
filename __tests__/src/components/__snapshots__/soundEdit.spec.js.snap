// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoundEdit/> render 1`] = `
<div
  className="item editing"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <input
        autoComplete="off"
        defaultValue="wind"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <label
        htmlFor="name"
      >
        <span>
          Name
        </span>
      </label>
      <span
        className="highlight"
      />
      <span
        className="bar"
      />
    </div>
    <div
      className="group"
    >
      <input
        autoComplete="off"
        defaultValue="wind"
        name="tags"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
      />
      <label
        htmlFor="tags"
      >
        <span>
          Tag
        </span>
      </label>
      <span
        className="highlight"
      />
      <span
        className="bar"
      />
    </div>
    <a
      className="button"
      onClick={[Function]}
      role="link"
      tabIndex={-1}
    >
      <span>
        Cancel
      </span>
    </a>
    <button
      className="button"
      style={
        Object {
          "backgroundColor": "#4CAF50",
          "borderColor": "#4CAF50",
        }
      }
      type="button"
    >
      <span>
        Save
      </span>
    </button>
  </form>
</div>
`;
